### DISCUSSION TITLE
What does Depth Resolution do?

### DISCUSSION THREAD

**Discussion by MaryAtlas** [nagadomi/nunif#168] [Q&A] [answered]
On iw3 GUI, there is Depth Resolution with Default option and 512. What is the value of Default? And which is better? What does it do?

**Comment by diverswan9** [comment]
This is coming from a non-expert here. And there's probably somewhere here that explains it. Maybe try searching in "Discussions" for 'Depth Resolution'. It has something to do with Depth Map sampling. Theoretically Higher resolution = better results. I found this from one of the posts, "DepthAnything is trained with 518 resolution, so I can guess that 518 is the best safe resolution." - (from nagadomi). You can enter your own numbers in there. I just put '518' in and leave it. Whatever number you enter in there needs to be divisible by 14 I think. The default is somewhere in the 300's I cant remember.

**Reply by MaryAtlas** [reply]
Alright, so back to my questions, you answered that Default is in the 300's. And are you saying the GUI 512 would be better than the default? If Higher resolution = better results, then there are no benefits in choosing Default since it is lower than the 518 that is used to train with, correct?
I have a follow up question to your answer, which is:
If 518 is the best, why nagadomi didn't put it in the GUI options? There must be a good reason for it?

**Reply by nagadomi** [reply]
Internally, the input size is converted to a valid size (e.g., multiple of 14 for DepthAnything, multiple of 32 for ZoeDepth).
With DepthAnything model, if 512 is specified, 518 is actually used. Because 512 is an invalid size that cannot be divided by 14.

The default value for ZoeDepth is 384 and the default value for DepthAnything is 392.
The default value for DepthAnything is the proper resolution for V1 Metric Depth model. It is also close to the ZoeDepth value.
518 is probably the best value for DepthAnything V1 relative depth model and V2, but the higher resolution means slower processing speed.
So, the default value adopts faster processing than depth quality.

**Reply by MaryAtlas** [reply]
Thank you. These are best answers from you.
One little part I am confused about. "The default value for DepthAnything is the proper resolution for V1 Metric Depth model" "518 is probably the best value for DepthAnything V1 relative depth model"
What do you mean by proper? Is it the best? Say I use ZeoD_Any_N, and I don't need faster processing, while I want the most accurate depth. Then generally would I use the proper resolution(Default?) or 518, or even higher?

**Reply by nagadomi** [reply, marked as answer]
392 is the size used for evaluation and training in the official code of DepthAnything V1 metric depth model (ZoeD_Any_N).
518 is the size used for evaluation and training in the official code of DepthAnything V1 relative depth model(Any_S,B,L) and all V2 models.

As for ZoeD_Any_N, I have no definite opinion about which is better, 518 or 392.
In general, the higher the resolution, the more detailed the depth. However, there is a problem that if resolution is too high, the depth will be broken. https://github.com/nagadomi/nunif/discussions/141
518 is not that high resolution, so probably not a problem.

**Reply by MaryAtlas** [reply]
Thank you!
